src/components/wrapper/song-list-wrapper.tsx

```diff
@@ -0,0 +1,158 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react"
+import { PlusIcon } from "@heroicons/react/24/solid"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { Suggestion } from "@/types/database-types"
+import { MagnifyingGlassCircleIcon } from "@heroicons/react/24/outline"
+import Spinner from "@/components/utils/spinner"
+import ErrorPopup from "@/components/popups/error-popup"
+import { useRouter } from "next/router"
+import { MagnifyingGlassIcon } from "@heroicons/react/20/solid"
+import SearchBar from "@/components/suggestion/search-bar"
+import { getRepertoireSongs, getSuggestions } from "@/services/suggestion.service"
+
+interface SongListPageWrapperProps {
+  renderSongCard: (suggestion: Suggestion) => JSX.Element
+  pageName: string
+}
```

Comment:
Perhaps just use an enum for this 

```ts
export enum SongType {
  Repertoire,
  Suggestion
}
```

You can then set the songs based on the enum like this:
```ts
const songs = songType == SongType.Suggestions 
              ? getSuggestions(supabaseClient) 
              : getRepertoireSongs(supabaseClient)
```

And then set the songCard based on the SongType in a similar way. 

This is less prone to errors (for example, a type in the pageName could be easily made) and gives us auto completion.